client/src/mapper-consents/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { apiErrorMessage, getMapperConsents, useApi } from '../api';
+import { BeatmapInline } from '../BeatmapInline';
+import { IMapperConsent } from '../interfaces';
+import { UserInline } from '../UserInline';
+
+function consentToCell(consent?: 0 | 1 | 2 | boolean) {
+  let index: 'null' | 0 | 1 | 2;
+  if (consent === true) {
+    index = 1
+  } else if (consent === false) {
+    index = 0
+  } else {
+    index = consent == null ? 'null' : consent;
+  }
+  const mapping = {
+    'null': ['no reply', 'pending'],
+    0: ['no', 'error'],
+    1: ['yes', 'success'],
+    2: ['unreachable', 'pending']
+  };
+  const [consentStr, className] = mapping[index];
+
+  return (
+    <td className={className}>
+      {consentStr}
+    </td>
+  )
+}
+
+function maybeBeatmapsetTable(consent: IMapperConsent) {
+  if (consent.beatmapset_consents.length === 0) {
+    return
+  }
+  return (
+    <tr key={consent.id + "-dropdown"}>
+      <td colSpan={3}>{MapperBeatmapsetConsents(consent)}</td>
+    </tr>
+  )
+}
+
+function MapperBeatmapsetConsents(mapperConsent: IMapperConsent) {
+  return (
+    <table style={{'width': '80%', 'marginLeft': '3em', 'marginRight': '3em', 'tableLayout': 'fixed'}}>
+      <tr>
+          <th style={{'width': '30%'}}>Beatmapset</th>
+          <th style={{'width': '10%'}}>Consent</th>
+          <th style={{'width': '60%'}}>Notes</th>
+      </tr>
+      {mapperConsent.beatmapset_consents.map((consent) => {
+        // TODO remove when beatmapset foreign key issues have been resolved
+        if (consent.beatmapset == null) {
+          return null;
+        }
+        return (
+          <>
+            <tr key={mapperConsent.id + "-beatmapset-" + consent.beatmapset.id}>
+              <td><BeatmapInline beatmapset={consent.beatmapset!} /></td>
+              {consentToCell(consent.consent)}
+              <td>{consent.consent_reason}</td>
+            </tr>
+          </>
+        )
+      })}
+    </table>
+  )
+}
+
+export default function MapperConsents() {
+  const [consents, consentError] = useApi(getMapperConsents);
+
+  if (consentError != null)
+    return <span className='panic'>Failed to load mapper consents: {apiErrorMessage(consentError)}</span>;
+
+  if (consents == null)
+    return <span>Loading mapper consents...</span>;
```

Comment:
this isnt something u need to fix but just letting u know the reason that other components do something like
```tsx
<div ...>
  <h1>...</h1>
  <some other component that calls api>
</div>
```
is so that the error message goes inside the document and after the header

it doesn't really matter though. this method of error handling is pretty dumb to begin with

Reply:
gotcha I see my mistake here